app/models/manageiq/providers/azure/cloud_manager/event_target_parser.rb

```diff
@@ -31,30 +31,40 @@ def parse_event_target(target_collection, event_data)
     resource_id   = event_data.try(:[], "resourceId")
     resource_type = event_data.try(:[], "resourceType").try(:[], "value")
     association   = case resource_type
-                    when "Microsoft.Network/networkSecurityGroups"
+                    when "Microsoft.Network/networkSecurityGroups", "Microsoft.Network/networkSecurityGroups/securityRules"
                       :security_groups
                     when "Microsoft.Network/networkInterfaces"
                       :network_ports
-                    when "Microsoft.Compute/virtualMachines"
+                    when "Microsoft.Compute/virtualMachines", "Microsoft.Authorization/locks"
                       :vms
                     when "Microsoft.Network/loadBalancers"
                       :load_balancers
-                    when "Microsoft.Network/publicIPAddresses"
+                    when /Microsoft.Network\/publicI[Pp]Addresses/
```

Comment:
You can just use the "i" switch instead of fiddling with character classes. And you may want to escape a literal dot between "Microsoft" and "Network", though it's not a huge deal here.

Reply:
Agreed with the dot. Though I'm not sure about the case-insensitive matching - It's a really annoying thingy this Microsoft miracle does. Really. In some cases, for some events (which fortunately we're not using - the commented out `deployments` events), this difference between lowercase and camelcase means also a difference in behavior - so our countermeasures are sometimes counterproductive. It's really a mess.

So I would rather prefer to stay with this precise matching which says: "Only this letter can be uppercase or lowercase and all other text must be precisely this, because here we're sure how it behaves." Or what do you thing?

Reply:
@tumido Where and why do we rely on the case of the resource type? That seems really strange. It shouldn't matter.

Reply:
@djberg96 So that is my original code, we do exact matches in the DB, so it's best to keep everything case sensitive. We transform this to ems_ref to the same format later, otherwise the targeted refresh would not work.

Btw. this is Azure's mess, they have resource_types documented right, but in events it's incorrect. Mostly in events of nested types.